cpp/core/compute/ExecutionCtx.h

```diff
@@ -78,54 +78,40 @@ class Backend : public std::enable_shared_from_this<Backend> {
     return substraitPlan_;
   }
 
-  virtual MemoryManager*
-  createMemoryManager(const std::string& name, std::shared_ptr<MemoryAllocator>, std::unique_ptr<AllocationListener>) {
-    throw GlutenException("createMemoryManager not implemented.");
-  }
+  virtual MemoryManager* createMemoryManager(
+      const std::string& name,
+      std::shared_ptr<MemoryAllocator>,
+      std::unique_ptr<AllocationListener>) = 0;
 
   /// This function is used to create certain converter from the format used by
   /// the backend to Spark unsafe row.
-  virtual std::shared_ptr<ColumnarToRowConverter> getColumnar2RowConverter(MemoryManager* memoryManager) {
-    throw GlutenException("getColumnar2RowConverter not implemented.");
-  }
+  virtual std::shared_ptr<ColumnarToRowConverter> getColumnar2RowConverter(MemoryManager* memoryManager) = 0;
 
   virtual std::shared_ptr<RowToColumnarConverter> getRowToColumnarConverter(
       MemoryManager* memoryManager,
-      struct ArrowSchema* cSchema) {
-    throw GlutenException("getRowToColumnarConverter not implemented.");
-  }
+      struct ArrowSchema* cSchema) = 0;
 
   virtual std::shared_ptr<ShuffleWriter> createShuffleWriter(
       int numPartitions,
       std::shared_ptr<ShuffleWriter::PartitionWriterCreator> partitionWriterCreator,
       const ShuffleWriterOptions& options,
-      MemoryManager* memoryManager) {
-    throw GlutenException("createShuffleWriter not implemented.");
-  }
+      MemoryManager* memoryManager) = 0;
 
-  virtual std::shared_ptr<Metrics> getMetrics(ColumnarBatchIterator* rawIter, int64_t exportNanos) {
-    return nullptr;
-  }
+  virtual std::shared_ptr<Metrics> getMetrics(ColumnarBatchIterator* rawIter, int64_t exportNanos) = 0;
```

Comment:
I'm fine with this change and as it already is. But I think in future, we should avoid touching unrelated code as far as possible to keep every pr minimum modification. Each pr should force on it's own topic to make easy to review, and history track.

Reply:
+1 on your opinion, however I think this change is needed, ExecutionCtx.h should be a pure abstract class as much as possible, old codes may due to history reason that CH backend has shared these codes before.